### PR TITLE
Add Megatron-Bridge recipe-free distillation example script

### PR DESCRIPTION
**Pull request by kevalmorabia97**
## What does this PR do?

**Type of change:** New example script 

- [x] M-Bridge recipe-free distillation script so its more easier to run and can support pruned models
- [x] Fix resuming distillation run

## Usage

```python
torchrun --nproc_per_node 8 distill.py \
    --teacher_hf_path Qwen/Qwen3-8B \
    --student_hf_path Qwen3-8B-NAS-Pruned-6B \
    --tp_size 8 \
    --data_paths <climbmix 25% tokenized (~90B tokens)> \
    --data_path_to_cache /path/to/cache/climbmix_dataset_indices_qwen3 \
    --seq_length 4096 \
    --mbs 8 \
    --gbs 768 \
    --train_iters 28500 \
    --lr 1e-4 \
    --min_lr 1e-5 \
    --lr_warmup_iters 100 \
    --eval_interval 500 \
    --eval_iters 32 \
    --log_interval 10 \
    --output_dir qwen3_8b_6b_mbridge_distill
```

## Testing

- [x] Re-ran Qwen3 8B -> 6B experiments and compare with Nemo2 results from blog

Best subnet from NAS: `{'num_layers': 30, 'hidden_size': 3584, 'ffn_hidden_size': 11776} -> 5.99B params, 0.5718 score`

| Model | MMLU | GSM8K - flexible, strict | MBPP (coding) |
| ------- | ------ | ------- | ------- |
| Qwen3-8B | 74.9 | 87.5, 84.6 | 65.4 |
| Qwen3-8B-Pruned-6B | 57.6 | 11.6, 10.0 | 4.8 |
| Qwen3-8B-Pruned-6B (Distilled for 16k steps i.e. 50B tokens ~3k GPU hours) | 71.6 | 78.0, 64.7 | 43.4 |
| Qwen3-8B-Pruned-6B (Distilled for 28.5k steps i.e. 90B tokens ~5.2k GPU hours) | 71.9 | 78.1, 64.8 | 44.2 |
| Qwen3-4B | 70.0 | 81.1, 84.7 | 62.8 |

Previous Nemo2 experiments on depth pruned Qwen3 8B -> 6B (24 layers) had MMLU ~72.0 so more or less similar. No hparam tuning done for current M-Bridge distillation run

- [ ] (Separate PR) GitHub CI/CD test for example script with NeMo 26.02 container

## Before your PR is "*Ready for review*"

- **Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/Model-Optimizer/blob/main/CONTRIBUTING.md)** and your commits are signed.
- **Is this change backward compatible?**: Yes 
- **Did you write any new necessary tests?**: N/A
- **Did you add or update any necessary documentation?**: Yes
- **Did you update [Changelog](https://github.com/NVIDIA/Model-Optimizer/blob/main/CHANGELOG.rst)?**: Yes 

## Release Notes

* **New Features**
  * Added complete distillation workflow and example for Megatron-Bridge optimization.

* **Documentation**
  * Enhanced setup guide with Docker workflows, data preparation steps, and detailed distillation instructions.
  * Improved usage documentation and help references.

* **Improvements**
  * Better data preprocessing output with human-readable formatting for metrics.

### CODE REVIEW

File: examples/megatron_bridge/distill.py [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+            ckpt_format="torch_dist",
+            fully_parallel_save=True,
+            finetune=True,
+        ),
+        rng=RNGConfig(seed=SEED),
+        mixed_precision="bf16_mixed",
+    )
+
+    print_rank_0("\nStarting distillation...")
+    distill(config)
```
Comment (AAnoosheh) [inline examples/megatron_bridge/distill.py:238]: Should we make it like the Nemo one where it can do either `pretrain()`, `distill()`, or `finetune()` all in one file?  (@ChenhanYu would that be preferred?)

File: examples/megatron_bridge/README.md [hunk 2]
```diff
@@ -50,7 +50,7 @@ torchrun --nproc_per_node 2 /opt/Megatron-Bridge/3rdparty/Model-Optimizer/exampl
 To see the full usage for advanced configurations, run:
 
 ```bash
-python /opt/Megatron-Bridge/3rdparty/Model-Optimizer/examples/megatron_bridge/prune_minitron.py --help
+torchrun --nproc_per_node 1 /opt/Megatron-Bridge/3rdparty/Model-Optimizer/examples/megatron_bridge/prune_minitron.py --help
```
Comment (AAnoosheh) [inline examples/megatron_bridge/README.md]: Why this?

Comment (kevalmorabia97) [reply examples/megatron_bridge/README.md]: I want to only print help on rank 0 so need to initialize multiprocesses which will only happen on torchrun. I am not spawning multiprocesses in the script so running with `python ...` will also result in an error trying to fine `RANK` env variable during dist setup

File: examples/megatron_bridge/distill.py (hunk 1 above)
Comment (kevalmorabia97) [reply examples/megatron_bridge/distill.py:238]: How about we incrementally extend this file as we get to needing these options?

Comment (kevalmorabia97) [reply examples/megatron_bridge/distill.py:238]: Maybe I should rename to `train.py`?

File: examples/megatron_bridge/README.md (hunk 2 above)
Comment (AAnoosheh) [reply examples/megatron_bridge/README.md]: mbridge has it's own `print_rank_0` which accounts for that

File: examples/megatron_bridge/distill.py (hunk 1 above)
Comment (AAnoosheh) [reply examples/megatron_bridge/distill.py:238]: I guess right now we can easily just put a `pretrain()` call if the KD-specific args aren't provided.
SFT can be done later since dataset/template/etc is different.

File: examples/megatron_bridge/README.md (hunk 2 above)
Comment (AAnoosheh) [reply examples/megatron_bridge/README.md]: We should additionally change our own `print_rank_0` to work without dist initialized

Comment (kevalmorabia97) [reply examples/megatron_bridge/README.md]: Our `print_rank_0` works fine on non-dist env. The issue here is I am manually doing `dist.setup()` which fails if not running with torchrun. Since I am doing all low-level M-bridge stuff myself because of lack of top-level APIs, we dont get the dist setup from M-Bridge

File: examples/megatron_bridge/distill.py [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+        max_lr=args.lr,
+        min_lr=args.min_lr,
+        adam_beta2=0.98,
+    )
+
+    # Build dataset config
+    dataset_kwargs = {
+        "seq_length": args.seq_length,
+        "path_to_cache": args.data_path_to_cache,
+        "random_seed": SEED,
```
Comment (jenchen13) [inline examples/megatron_bridge/distill.py:173]: can the seed also be a random arg?

File: examples/megatron_bridge/README.md [hunk 4]
```diff
@@ -4,21 +4,47 @@ This directory contains examples of using Model Optimizer with [NeMo Megatron-Br
...
+
+export DOCKER_IMAGE=nvcr.io/nvidia/nemo:26.02
+docker run \
+  --gpus all \
+  --shm-size=16GB \
+  --net=host \
+  --ulimit memlock=-1 \
+  --rm -it \
+  -v ${MODELOPT_DIR}:/opt/Model-Optimizer \
+  -v ${MODELOPT_DIR}/modelopt:/opt/venv/lib/python3.12/site-packages/modelopt \
```
Comment (jenchen13) [inline examples/megatron_bridge/README.md:37]: why is mounting to venv also necessary?

File: examples/megatron_bridge/README.md [hunk 5]
```diff
@@ -60,7 +88,102 @@ python /opt/Megatron-Bridge/3rdparty/Model-Optimizer/examples/megatron_bridge/pr
...
+### Slurm Usage
+
+To run the distillation script on a Slurm cluster for multi-node training, you just need use `python` instead of `torchrun` and set the number of nodes using `#SBATCH --nodes=<num_nodes>` clause in your Slurm script.
+
+### Convert Megatron checkpoint to Hugging Face format
+
+To convert the Megatron checkpoint from last iteration (or any intermediate iteration) to Hugging Face format, you need the pruned model config (`--output_hf_path` from `prune_minitron.py` script) and the distilled megatron checkpoint dir (`<distill_output_dir>/checkpoints/iter_<iter_number>`) to run the following command:
+
+```bash
+uv run python /opt/Megatron-Bridge/examples/conversion/convert_checkpoints.py export \
```
Comment (jenchen13) [inline examples/megatron_bridge/README.md:180]: do we assume the user already has uv installed?

Comment (kevalmorabia97) [reply examples/megatron_bridge/README.md:180]: Its in the nemo container so already installed

File: examples/megatron_bridge/README.md (hunk 4 above)
Comment (kevalmorabia97) [reply examples/megatron_bridge/README.md:37]: So users can mount library and examples from same version. This avoids the case where user uses old modelopt but with examples from main branch

File: examples/megatron_bridge/distill.py (hunk 3 above)
Comment (kevalmorabia97) [reply examples/megatron_bridge/distill.py:173]: You mean randomly generate everytime? Then the results may not be reproducible
